compiler/rustc_ast_lowering/src/path.rs

```diff
@@ -111,11 +111,9 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
                     )
                 },
             )),
-            span: self.lower_span(
-                p.segments[..proj_start]
-                    .last()
-                    .map_or(path_span_lo, |segment| path_span_lo.to(segment.span())),
-            ),
+            span: p.segments[..proj_start]
+                .last()
+                .map_or(path_span_lo, |segment| path_span_lo.to(segment.span())),
```

Comment:
It would appear that half the regression in incr-unchecked comes from `Span::to` in this line. The other half comes from using a MutVisitor in DefCollector, which had to be expected.

Reply:
>The other half comes from using a MutVisitor in DefCollector

I tried to address this in https://github.com/rust-lang/rust/pull/127371, not sure if it will help.